### ISSUE TITLE
"lookupSymbol failed in relocateSection" error when running `hie-bios check` on macOS

### ISSUE THREAD

**Issue by armatures** [mpickering/hie-bios#303]
I pulled the latest commits in this repo, ran `cabal install` to get the latest `hie-bios`. I have a very simple "hello world" project using [the latest LTS package set](https://www.stackage.org/lts-18.4) with a simple stack `hie.yaml`.

In that project, running `hie-bios check .` gives the following error:
```
hie-bios:
lookupSymbol failed in relocateSection (relocate external)
/Users/me/.stack/programs/x86_64-osx/ghc-8.10.4/lib/ghc-8.10.4/ghc-prim-0.6.1/HSghc-prim-0.6.1.o: unknown symbol `_registerForeignExports'
hie-bios: hie-bios: unable to load package `ghc-prim-0.6.1'
```
running `hie-bios debug .` gives:
```
Root directory:        /Users/me/Exercism/haskell/hello-world
Component directory:   /Users/me/Exercism/haskell/hello-world
GHC options:           -i -odir=/Users/me/Exercism/haskell/hello-world/.stack-work/odir -hidir=/Users/me/Exercism/haskell/hello-world/.stack-work/odir -hide-all-packages -i/Users/me/Exercism/haskell/hello-world/.stack-work/dist/x86_64-osx/Cabal-3.2.1.0/build -i/Users/me/Exercism/haskell/hello-world/src -i/Users/me/Exercism/haskell/hello-world/.stack-work/dist/x86_64-osx/Cabal-3.2.1.0/build/autogen -i/Users/me/Exercism/haskell/hello-world/.stack-work/dist/x86_64-osx/Cabal-3.2.1.0/build/global-autogen -stubdir=/Users/me/Exercism/haskell/hello-world/.stack-work/dist/x86_64-osx/Cabal-3.2.1.0/build -package-id=base-4.14.1.0 -Wall -optP-include -optP/Users/me/Exercism/haskell/hello-world/.stack-work/ghci/a961ff61/cabal_macros.h -ghci-script=/private/var/folders/kw/c01ws73s1xn996t2c_jt_fr00000gn/T/haskell-stack-ghci/f4670df3/ghci-script -package-db /Users/me/Exercism/haskell/hello-world/.stack-work/install/x86_64-osx/cadb96a673cf8628d6f664c88359dded38d273d859766adb81741ca234047130/8.10.4/pkgdb -package-db /Users/me/.stack/snapshots/x86_64-osx/cadb96a673cf8628d6f664c88359dded38d273d859766adb81741ca234047130/8.10.4/pkgdb -package-db /Users/me/.stack/programs/x86_64-osx/ghc-8.10.4/lib/ghc-8.10.4/package.conf.d
GHC library directory: CradleSuccess "/Users/me/.stack/programs/x86_64-osx/ghc-8.10.4/lib/ghc-8.10.4"
GHC version:           CradleSuccess "8.10.4"
Config Location:       No explicit config found
Cradle:                Cradle {cradleRootDir = "/Users/me/Exercism/haskell", cradleOptsProg = CradleAction: Default}
Dependencies:          hello-world.cabal package.yaml stack.yaml
```

Thank you!

**Comment by fendor** [comment]
Hi, just to make sure, you compiled hie-bios with the same ghc version, e.g. `8.10.4`?

The `CradleAction` looks weird, it should be named `Stack`... But the ghc options look like something a stack cradle would have generated.

**Comment by armatures** [comment]
That fixed it! Thank you and sorry for the trouble!

Running `hie-bios --help` gives the version I compiled with, and it was `8.10.1`. I followed [these instructions](https://www.haskell.org/downloads/#linux-mac-freebsd) to get [GHCup](https://www.haskell.org/ghcup/), ran `ghcup  tui` and chose `8.10.4`. To reinstall `hie-bios` built with the right version, I ran `cabal install --overwrite-policy=always`. After that, the `check` command returned an exit code of 0 👌.
